engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/keys/ScatterKey.scala

```diff
@@ -89,17 +96,33 @@ private [execution] case class ScatterKey(node: ScatterNode) extends JobKey {
       executionStoreChanges.toMap -> valueStoreChanges.toMap
     }
 
+    // Checks the scatter width of a scatter node and builds WorkflowExecutionDiff accordingly
+    // If scatter width is more than max allowed limit, it fails the ScatterNode key
+    def buildExecutionDiff(scatterSize: Int, arrays: List[ScatterVariableAndValue]): WorkflowExecutionDiff = {
+      if(scatterSize > MaxScatterWidth) {
+        workflowExecutionActor ! JobFailedNonRetryableResponse(this, new Exception(s"Workflow has scatter width $scatterSize, which is more than the max scatter width $MaxScatterWidth allowed per scatter!"), None)
+
+        // even though the status is updated to Running, it will eventually be changed to Failed when WEA receives JobFailedNonRetryableResponse event
+        WorkflowExecutionDiff(Map(this -> ExecutionStatus.Running))
```

Comment:
Is there harm in setting this to Failed instead? It might be kind of confusing to debug this (not that it would ever need debugging of course 😛 ) if one sees the job in Running state when a job is started after the execution limit has been hit.

Reply:
I don't see any harm in setting it to Failed state. Will make that change. 👍